aspnetcore/tutorials/min-web-api.md

```diff
@@ -0,0 +1,384 @@
+-->
+
+---
+
+`Hello World!` is displayed in the browser.
+
+## Add NuGet packages
+
+NuGet packages must be added to support the database and diagnostics used in this tutorial.
+
+# [Visual Studio](#tab/visual-studio)
+
+* From the **Tools** menu, select **NuGet Package Manager > Manage NuGet Packages for Solution**.
+* Select the **Browse** tab, and verify that **Include prerelease** is checked.
+* Enter **Microsoft.EntityFrameworkCore.InMemory** in the search box, and then select `Microsoft.EntityFrameworkCore.InMemory`.
+* Select the **Project** checkbox in the right pane and then select **Install**.
+* Follow the preceding instructions to add the `Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore` package.
+
+# [Visual Studio Code](#tab/visual-studio-code)
+
+* Run the following commands:
+
+   ```dotnetcli
+   dotnet add package Microsoft.EntityFrameworkCore.InMemory --prerelease
+   dotnet add package Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore --prerelease
+     ```
+
+<!-- add VS Mac later 
+# [Visual Studio for Mac](#tab/visual-studio-mac)
+
+Open a command terminal in the project folder and run the following command:
+
+   ```dotnetcli
+   dotnet add package Microsoft.EntityFrameworkCore.InMemory
+   ```
+   -->
+---
+
+## Add the API code
+
+Replace the contents of the *program.cs* file with the following sample app code:
+
+[!code-csharp[](min-web-api/samples/6.x/todo/Program.cs?name=snippet_all)]
+
+## The model and database context classes
+
+The sample app contains the following model:
+
+[!code-csharp[](min-web-api/samples/6.x/todo/Program.cs?name=snippet_model)]
+
+A *model* is a set of classes that represent the data that the app manages. The model for this app is the `Todo` class.
+
+The sample app also contains the following database context class:
+
+[!code-csharp[](min-web-api/samples/6.x/todo/Program.cs?name=snippet_cntx)]
+
+The *database context* is the main class that coordinates [Entity Framework](/ef/core/) functionality for a data model. This class is created by deriving from the <xref:Microsoft.EntityFrameworkCore.DbContext?displayProperty=fullName> class.
+
+The following highlighted code adds the database context to the [dependency injection (DI)](xref:fundamentals/dependency-injection) container:
+
+[!code-csharp[](min-web-api/samples/6.x/todo/Program.cs?name=snippet_DI&highlight=3)]
+
+ The DI container provides access to the database context and other services.
+
+## Install Postman to test the app
+
+This tutorial uses Postman to test the API.
+
+* Install [Postman](https://www.getpostman.com/downloads/)
+* Start the web app.
+* Start Postman.
+* Disable **SSL certificate verification**
+  * From **File** > **Settings** (**General** tab), disable **SSL certificate verification**.
+    > [!WARNING]
+    > Re-enable SSL certificate verification after testing the controller.
+
+<a name="post"></a>
+
+### Test posting data
```

Comment:
If we used Web API template, you'd have swaggerUI in there as well which provides the default testing place as well...seems like Postman isn't an ideal here IMO.

Reply:
Per previous comment, I'd like to avoid swagger to start with and let them add Swagger in another tutorial.

Reply:
What I'm trying to better understand is shouldn't a tutorial END in the same spot that our tools actually produce? We have a template that bootstraps a lot of this for a reason, to provide a good starting point we believe in for APIs...otherwise why would we have created a starting point for it?

Reply:
That's a [different API tutorial](https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?view=aspnetcore-5.0&tabs=visual-studio) , the controller based API.  The tradditonal Web API template generates the following code:

```
using Microsoft.OpenApi.Models;

var builder = WebApplication.CreateBuilder(args);

// Add services to the container.

builder.Services.AddControllers();
builder.Services.AddSwaggerGen(c =>
{
    c.SwaggerDoc("v1", new() { Title = "WebAPI6", Version = "v1" });
});

var app = builder.Build();

// Configure the HTTP request pipeline.
if (builder.Environment.IsDevelopment())
{
    app.UseDeveloperExceptionPage();
    app.UseSwagger();
    app.UseSwaggerUI(c => c.SwaggerEndpoint("/swagger/v1/swagger.json", "WebAPI6 v1"));
}

app.UseHttpsRedirection();

app.UseAuthorization();

app.MapControllers();

app.Run();
```

That's not a good starting point for `MapGet` , `MapPost`, etc. I'd have to explain controllers , `app.MapControllers();` and then tell them why we're deleting that code.

Reply:
This feels weird to me TBH...